tests/WithProgress.hs

```diff
@@ -0,0 +1,10 @@
+import Test.QuickCheck
+import Test.QuickCheck.Monadic
+
+main = quickCheck prop_test_withProgress
```

Comment:
I would really appreciate an actual test here that maybe checks that you get reasonable numbers out of your `TestProgress` datastructure. You could dump the numbers to `IORef`s or similar and fail the test if the numbers we get don't make much sense.

Reply:
Sure, that makes more sense. Now I performed the very scientific 'ocular inspection' to assess the correctness!